### PR TITLE
Update comments

### PR DESCRIPTION
**Pull request by victorwiebe**
Updated comments in this PR
Also added return types for call async methods
Also added small fixes (i.e. changing BN to numbers) along the way

### CODE REVIEW

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 1]
```diff
@@ -1014,46 +1213,95 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
...
+   */
+  public owner = async (): Promise<string> => {
     return (await this.contract).owner.callAsync();
   };
 
-  public isIssuable = async () => {
+  /**
+   * A security token issuer can specify that issuance has finished for the token
+   * (i.e. no new tokens can be minted or issued).
+   * @dev If a token returns FALSE for `isIssuable()` then it MUST always return FALSE in the future.
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tags

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 2]
```diff
@@ -1014,46 +1213,95 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
...
+   * @return true if `msg.sender` is the owner of the contract.
+   */
   public isOwner = async (): Promise<boolean> => {
     return (await this.contract).isOwner.callAsync();
   };
 
+  /**
+   * In order to provide transparency over whether `controllerTransfer` / `controllerRedeem` are useable
+   * or not `isControllable` function will be used.
+   * @dev If `isControllable` returns `false` then it always return `false` and
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tags

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 3]
```diff
@@ -1117,11 +1384,19 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
...
+   * @return address List of modules with this name
+   */
   public getModulesByName = async (params: ModuleNameParams) => {
     const moduleNameHex = stringToBytes32(params.moduleName);
     return (await this.contract).getModulesByName.callAsync(moduleNameHex);
   };
 
+  /**
+   * Allows the owner to withdraw unspent POLY stored by them on the ST or any ERC20 token.
+   * @dev Owner can transfer POLY to the ST which will be used to pay for modules that require a POLY fee.
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tags

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 4]
```diff
@@ -1191,49 +1484,82 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
-  public getInvestors = async () => {
+  /**
+   * returns an array of investors
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Capital letter (same for next three)

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 5]
```diff
@@ -1274,6 +1608,12 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * This function must be called to increase the total supply (Corresponds to mint function of ERC20).
+   * @dev It is only be called by the token issuer or the operator defined by the issuer. ERC1594 doesn't have
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tag

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 6]
```diff
@@ -1327,6 +1674,12 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * Validate permissions with PermissionManager if it exists, If no Permission return false
+   * @dev Note that IModule withPerm will allow ST owner all permissions anyway
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tags

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 7]
```diff
@@ -1359,6 +1720,15 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * Decreases totalSupply and the corresponding amount of the specified partition of tokenHolder
+   * @dev This function can only be called by the authorised operator.
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove all tags (params should be covered in the interface)

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 8]
```diff
@@ -1375,6 +1745,12 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * This function redeem an amount of the token of a msg.sender. For doing so msg.sender may incentivize
+   * using different ways that could be implemented with in the `redeem` function definition. But those implementations
+   * are out of the scope of the ERC1594.
+   * @dev It is analogy to `transferFrom`
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tags

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 9]
```diff
@@ -1521,6 +1944,10 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * Used by the issuer to permanently disable controller functionality
+   * @dev enabled via feature switch "disableControllerAllowed"
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tag

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 10]
```diff
@@ -1531,6 +1958,12 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * This function allows an authorised address to transfer tokens between any two token holders.
+   * The transfer must still respect the balances of the token holders (so the transfer must be for at most
+   * `balanceOf(_from)` tokens) and potentially also need to respect other transfer restrictions.
+   * @dev This function can only be executed by the `controller` address.
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tag

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 11]
```diff
@@ -1548,6 +1981,12 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * This function allows an authorised address to redeem tokens for any token holder.
+   * The redemption must still respect the balances of the token holder (so the redemption must be for at most
+   * `balanceOf(tokenHolder)` tokens) and potentially also need to respect other transfer restrictions.
+   * @dev This function can only be executed by the `controller` address.
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tag

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 12]
```diff
@@ -1574,10 +2016,21 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     return (await this.contract).getModulesByType.callAsync(params.type);
   };
 
+  /**
+   * use to return the global treasury wallet
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Capital letter

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 13]
```diff
@@ -1574,10 +2016,21 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
...
+   */
   public getTreasuryWallet = async (): Promise<string> => {
     return (await this.contract).getTreasuryWallet.callAsync();
   };
 
+  /**
+   * Function used to attach a module to the security token
+   * @dev  E.G.: On deployment (through the STR) ST gets a TransferManager module attached to it
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tags

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 14]
```diff
@@ -1591,6 +2044,11 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * Attachs a module to the SecurityToken
+   * @dev  E.G.: On deployment (through the STR) ST gets a TransferManager module attached to it
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tags

Also, should be *Attaches*

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 15]
```diff
@@ -1747,6 +2212,10 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     return typedResult;
   };
 
+  /**
+   * Used to attach a new document to the contract, or update the URI or hash of an existing attached document
+   * @dev Can only be executed by the owner of the contract.
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tag

File: src/contract_wrappers/tokens/security_token_wrapper.ts [hunk 16]
```diff
@@ -1760,6 +2229,10 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  /**
+   * Used to remove an existing document from the contract by giving the name of the document.
+   * @dev Can only be executed by the owner of the contract.
```
Comment (monitz87) [inline src/contract_wrappers/tokens/security_token_wrapper.ts]: Remove `@dev` tag

File: src/contract_wrappers/modules/checkpoint/dividend_checkpoint_wrapper.ts [hunk 17]
```diff
@@ -27,61 +27,109 @@ export namespace DividendCheckpointTransactionParams {
...
+/**
+ * @param dividendIndex Dividend to push
+ * @param start Index in investor list at which to start pushing dividends
+ * @param end Index in investor list at which to stop pushing dividends
+ */
 interface PushDividendPaymentParams extends TxParams {
   dividendIndex: number;
   start: Date;
-  iterations: number;
+  end: Date;
```
[the remark this answers is not in the document]
Comment (victorwiebe) [reply src/contract_wrappers/modules/checkpoint/dividend_checkpoint_wrapper.ts]: Checkpoint indexes?

[the remark this answers is not in the document]
Comment (victorwiebe) [reply src/contract_wrappers/modules/checkpoint/dividend_checkpoint_wrapper.ts]: ` Index in investor list at which to stop pushing dividends`

File: src/contract_wrappers/modules/permission_manager/general_permission_manager_wrapper.ts [hunk 18]
```diff
@@ -306,8 +304,8 @@ export default class GeneralPermissionManagerWrapper extends ModuleWrapper {
 
   /**
    * Used to return all permission of a single or multiple module
-   * @return address[] the address array of Modules this delegate has permission
-   * @return bytes32[] the permission array of the corresponding Modules
+   * @return address[] the address array of Modules this delegate has permission, bytes32[]
```
Comment (monitz87) [inline src/contract_wrappers/modules/permission_manager/general_permission_manager_wrapper.ts]: Change `address` and `bytes32` to the corresponding types

File: src/contract_wrappers/modules/sto/capped_sto_wrapper.ts [hunk 19]
```diff
@@ -301,15 +301,11 @@ export default class CappedSTOWrapper extends STOWrapper {
...
-   * @return Date at which offering gets start.
-   * @return Date at which offering ends.
-   * @return Number of token base units this STO will be allowed to sell to investors.
-   * @return Token units a buyer gets as the rate
-   * @return Amount of funds raised
-   * @return Number of individual investors this STO have.
-   * @return Amount of tokens get sold.
-   * @return Boolean value to justify whether the fund raise type is POLY or not, i.e true for POLY.
-   * @return Boolean value to know the nature of the STO Whether it is pre-mint or mint on buying type sto.
+   * @return Date at which offering gets start., Date at which offering ends., Number of token base units this STO will
```
Comment (monitz87) [inline src/contract_wrappers/modules/sto/capped_sto_wrapper.ts]: Remove extra `.`s

File: src/contract_wrappers/modules/sto/usd_tiered_sto_wrapper.ts [hunk 20]
```diff
@@ -734,15 +734,10 @@ export default class USDTieredSTOWrapper extends STOWrapper {
...
-   * @return Unixtimestamp at which offering gets start.
-   * @return Unixtimestamp at which offering ends.
-   * @return Currently active tier
-   * @return Array of Number of tokens this STO will be allowed to sell at different tiers.
-   * @return Array rate at which tokens are sold at different tiers
-   * @return Amount of funds raised
-   * @return Number of individual investors this STO have.
-   * @return Amount of tokens sold.
-   * @return Array of booleans to show if funding is allowed in ETH, POLY, SC respectively
+   * @return Unixtimestamp at which offering gets start., Unixtimestamp at which offering ends., Currently active tier,
```
Comment (monitz87) [inline src/contract_wrappers/modules/sto/usd_tiered_sto_wrapper.ts]: Remove extra `.`s

**Pull request merged**